### PR TITLE
draft: add bucket switch and sync with tikv

### LINKED ISSUE #4720: pd should support the switch of the buckets
# feature enhance
pd should support the switch of the buckets, and notice the tikv node by PutStore and StoreHeartBeat.

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?
PD should take control of the switch of the buckets and will be sync by PutStore and StoreHeartBeat.

Issue Number: Close #4720
depences on: https://github.com/tikv/pd/issues/4720

### What is changed and how it works?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

Code changes

- Has configuration change

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -1102,6 +1104,8 @@ type PDServerConfig struct {
 	TraceRegionFlow bool `toml:"trace-region-flow" json:"trace-region-flow,string,omitempty"`
 	// FlowRoundByDigit used to discretization processing flow information.
 	FlowRoundByDigit int `toml:"flow-round-by-digit" json:"flow-round-by-digit"`
+	// UseBucket is the switch to use bucket.
+	UseBucket bool `toml:"use-bucket" json:"use-bucket,string"`
```
Comment (nolouch) [inline server/config/config.go:1108]: Better use `EnableRegionBuckets`,`DisableRegionBuckets` or `EnableDynamicRegion`,`DisableDynamicRegion`. CC @BusyJay

Comment (rleungx) [reply server/config/config.go:1108]: `EnableXXX` is better.

Comment (bufferflies) [reply server/config/config.go]: fix

**Comment by bufferflies** [comment]
close #5013

**Pull request closed without merging**